### ISSUE TITLE
textual_inversion.py - Cannot set revision="fp16"

### ISSUE THREAD

**Issue by zagglez** [huggingface/diffusers#1543]
### Describe the bug

Errors with RuntimeError: Input type (float) and bias type (c10::Half) should be the same
if attempt to set revision="fp16"

### Reproduction

examples/textual_inversion.py errors if revision set to fp16

### Logs

```shell
Traceback (most recent call last):
  File "textual_inversion.py", line 659, in <module>
    main()
  File "textual_inversion.py", line 555, in main
    latents = vae.encode(batch["pixel_values"]).latent_dist.sample().detach()
  File "/opt/conda/lib/python3.8/site-packages/diffusers/models/vae.py", line 570, in encode
    h = self.encoder(x)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/diffusers/models/vae.py", line 130, in forward
    sample = self.conv_in(sample)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/conv.py", line 463, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/conv.py", line 459, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: Input type (float) and bias type (c10::Half) should be the same
```

### System Info

- `diffusers` version: 0.9.0
- Platform: Linux-5.15.0-56-generic-x86_64-with-glibc2.10
- Python version: 3.8.13
- PyTorch version (GPU?): 1.13.0+cu117 (True)
- Huggingface_hub version: 0.11.1
- Transformers version: 4.24.0

**Comment by patrickvonplaten** [comment]
@williamberman @patil-suraj could one of you take a look here?

**Comment by williamberman** [comment]
Looks like we can enable auto mixed precision training for this. Working on a PR

**Comment by patil-suraj** [comment]
This should be fixed with #1465, also in general for mixed precision training it's not recommended to use `fp16` weights as it could create instabilities during training. So I would recommend to not use `fp16` revision for training. See related issue which has more details about this  #1246

**Comment by williamberman** [comment]
closing this issue as it's covered by https://github.com/huggingface/diffusers/issues/1246 !
